From 05b926c75067eed23b85df2f717d8d9f255709cb Mon Sep 17 00:00:00 2001
From: Andrew Wang <wang@apache.org>
Date: Tue, 29 Mar 2016 11:54:42 -0700
Subject: [PATCH 1335/2748] HDFS-10197. TestFsDatasetCache failing
 intermittently due to timeout. Contributed by Lin
 Yiqun.

(cherry picked from commit f2aec4eb824647a01e14b4eede03af0babe65fb6)

Change-Id: If74d5e952d18eaf3c7b03b62df489846fd353eca
---
 .../java/org/apache/hadoop/hdfs/DFSTestUtil.java   |    2 +-
 .../hdfs/server/datanode/TestFsDatasetCache.java   |    5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/DFSTestUtil.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/DFSTestUtil.java
index a998efc..f5801a9 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/DFSTestUtil.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/DFSTestUtil.java
@@ -1393,7 +1393,7 @@ public Boolean get() {
             NativeIO.POSIX.getCacheManipulator().getMemlockLimit());
         return true;
       }
-    }, 100, 60000);
+    }, 100, 120000);
     return expectedCacheUsed;
   }
 
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestFsDatasetCache.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestFsDatasetCache.java
index c302324..2cd5cdc 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestFsDatasetCache.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/TestFsDatasetCache.java
@@ -88,6 +88,8 @@
 import com.google.common.base.Supplier;
 import com.google.common.primitives.Ints;
 
+import static org.apache.hadoop.hdfs.DFSConfigKeys.DFS_DATANODE_FSDATASETCACHE_MAX_THREADS_PER_VOLUME_KEY;
+
 public class TestFsDatasetCache {
   private static final Log LOG = LogFactory.getLog(TestFsDatasetCache.class);
 
@@ -124,6 +126,7 @@ public void setUp() throws Exception {
     conf.setLong(DFSConfigKeys.DFS_DATANODE_MAX_LOCKED_MEMORY_KEY,
         CACHE_CAPACITY);
     conf.setLong(DFSConfigKeys.DFS_HEARTBEAT_INTERVAL_KEY, 1);
+    conf.setInt(DFS_DATANODE_FSDATASETCACHE_MAX_THREADS_PER_VOLUME_KEY, 10);
 
     prevCacheManipulator = NativeIO.POSIX.getCacheManipulator();
     NativeIO.POSIX.setCacheManipulator(new NoMlockCacheManipulator());
@@ -452,7 +455,7 @@ public Boolean get() {
     }, 100, 10000);
   }
 
-  @Test(timeout=60000)
+  @Test(timeout=600000)
   public void testPageRounder() throws Exception {
     // Write a small file
     Path fileName = new Path("/testPageRounder");
-- 
1.7.9.5

